quinn-proto/src/token.rs

```diff
@@ -67,20 +67,30 @@ impl RetryToken {
             issued,
         })
     }
+}
 
-    fn put_additional_data<'b>(address: &SocketAddr, additional_data: &'b mut [u8]) -> &'b [u8] {
-        let mut cursor = &mut *additional_data;
-        match address.ip() {
-            IpAddr::V4(x) => cursor.put_slice(&x.octets()),
-            IpAddr::V6(x) => cursor.put_slice(&x.octets()),
+fn encode_addr(buf: &mut Vec<u8>, address: &SocketAddr) {
```

Comment:
Should this be an `impl coding::Codec` for `SocketAddr` instead?

Reply:
I considered that, but opted not to because:
- there's only one callsite
- the type tag makes serialization fallible for reasons other than length, which can't be expressed in `Codec`'s existing API.